Source/items.cpp

```diff
@@ -3214,17 +3225,21 @@ bool ItemSpaceOk(Point position)
 	if (dItem[position.x][position.y] != 0)
 		return false;
 
-	if (dObject[position.x][position.y] != 0) {
-		oi = dObject[position.x][position.y] > 0 ? dObject[position.x][position.y] - 1 : -(dObject[position.x][position.y] + 1);
-		if (Objects[oi]._oSolidFlag)
+	int oi = dObject[position.x][position.y];
+
+	if (oi != 0) {
+		if (Objects[abs(oi) - 1]._oSolidFlag)
 			return false;
 	}
 
-	if (dObject[position.x + 1][position.y + 1] > 0 && Objects[dObject[position.x + 1][position.y + 1] - 1]._oSelFlag != 0)
-		return false;
-
-	if (dObject[position.x + 1][position.y + 1] < 0 && Objects[-(dObject[position.x + 1][position.y + 1] + 1)]._oSelFlag != 0)
+	Point testPosition = position + DIR_S;
+	if (!InDungeonBounds(testPosition))
 		return false;
+	oi = dObject[testPosition.x][testPosition.y];
+	if (oi != 0) {
+		if (Objects[abs(oi) - 1]._oSelFlag != 0)
+			return false;
+	}
 
 	if (dObject[position.x + 1][position.y] > 0
 	    && dObject[position.x][position.y + 1] > 0

```

Comment:
This set of checks only looks at positive object indexes, not negative encoded ones like the previous checks. Not sure if this is a bug or what the reasoning behind it is.